src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -456,10 +480,7 @@ protected void ExecuteEvmCall(
 
             Snapshot snapshot = WorldState.TakeSnapshot();
 
-            // Fixes eth_estimateGas.
-            // If sender is SystemUser subtracting value will cause InsufficientBalanceException
-            if (validate || !tx.IsSystem())
-                WorldState.SubtractFromBalance(tx.SenderAddress, tx.Value, spec);
```

Comment:
`OriginalValidate` was introduced due to this condition. For every other condition validation off is equivalent with transaction being system transaction, but for this one condition we need to decouple this logic, and even if it is system transaction validation can be on/off independently, this is used in `eth_estimateGas`

Reply:
Ok, thanks. I understand the logic. On the other hand, this logic assumes that there are sysCalls that pay values. Based on my quick check, that’s not the case. In sysCalls, we set Value = 0. This means we could simplify the original check and remove that hacky edge case flag.

So original logic could be:
`if (validate && !tx.IsSystem())`